pkg/alexa/l10n/l10n.go

```diff
@@ -0,0 +1,246 @@
+	_, ok := r.locales[l.Name]
+	if ok {
+		return fmt.Errorf("locale %s already registered", l.Name)
+	}
+
+	// run all RegisterFuncs
+	var cfg Config
+	for _, opt := range opts {
+		opt(&cfg)
+	}
+
+	// order matters, first anything that can fail, then changing data
+	if cfg.FallbackFor != "" {
+		// fallback locale must be registered
+		orig, ok := r.locales[cfg.FallbackFor]
+		if !ok {
+			return fmt.Errorf("cannot be fallback for locale %s as it is not registered", cfg.FallbackFor)
+		}
+
+		// fallback already defined
+		if orig.Fallback != nil {
+			return fmt.Errorf("fallback for locale %s is already registered", orig.Name)
+		}
+
+		// set the fallback
+		r.locales[cfg.FallbackFor].Fallback = l
+	}
+
+	// set locale as default
+	if cfg.DefaultLocale {
+		r.defaultLocale = l.Name
+	}
+
+	r.locales[l.Name] = l
+
+	return nil
+}
+
+func (r *Registry) GetDefaultLocale() string {
+	return r.defaultLocale
+}
+func (r *Registry) GetLocales() map[string]*Locale {
+	return r.locales
+}
+
+// Resolve returns the Locale matching the given name or an error
+func (r Registry) Resolve(name string) (*Locale, error) {
+	l, ok := r.locales[name]
+	if !ok {
+		return nil, fmt.Errorf("locale %s not found", name)
+	}
+	return l, nil
+}
+
+// Get returns the translation for the snippet
+func (s Snippets) Get(k Key, args ...interface{}) (string, error) {
+	if len(s[k]) < 1 {
+		return "", fmt.Errorf("key not defined %s", string(k))
+	}
+	if len(s[k]) == 1 {
+		return fmt.Sprintf(s[k][0], args...), nil
+	}
+	l := len(s[k])
+	r := rand.Intn(l)
+	return fmt.Sprintf(s[k][r], args...), nil
+}
+
+func (s Snippets) GetAll(k Key, args ...interface{}) ([]string, error) {
+	if len(s[k]) < 1 {
+		return []string{}, fmt.Errorf("key not defined %s", string(k))
+	}
+	return s[k], nil
+}
+
+// GetSnippet returns the translation and follows fallback chain
+func (l Locale) GetSnippet(k Key, args ...interface{}) string {
+	if r, err := l.TextSnippets.Get(k, args...); err == nil {
+		return r
+	}
+	if l.Fallback != nil {
+		return l.Fallback.GetSnippet(k, args...)
+	}
+
+	return string(k)
+}
+
+// GetAllSnippets returns all available translations and follows fallback chain
+func (l Locale) GetAllSnippets(k Key, args ...interface{}) []string {
+	if r, err := l.TextSnippets.GetAll(k); err == nil {
+		return r
+	}
+	if l.Fallback != nil {
+		return l.Fallback.GetAllSnippets(k, args...)
+	}
+
+	return []string{string(k)}
+}
+
+func (l Locale) GetIntent(key Key) IntentResponse {
+	if r, ok := l.IntentResponses[key]; ok {
+		return r
+	}
+
+	if l.Fallback != nil {
+		return l.Fallback.GetIntent(key)
+	}
+
+	return IntentResponse{}
+}
+
+func (l Locale) GetSingleIntentResponse(key Key) (string, string, string) {
+	ir := l.GetIntent(key)
+
+	if len(ir.Text) > 0 {
+		r := rand.Intn(len(ir.Text))
+		return ir.Title[0], ir.Text[r], ir.SSML[r]
+	}
+
+	return string(key + ".Title"), string(key + ".Text"), string(key + ".SSML")
```

Comment:
Why are you adding text to the response? I dont understand what this function does anymore? Does it not get the responses. They look like keys.

Reply:
It does get the response, with fallback and eventually just the key + type string. 
I noticed that skill validation can fail with fallbacks, so I might remove the fallback altogether, especially for intents and real spoken text (that needs to be translated for the skill to pass validation).

`ask validate`:
> The skill's example phrases and description should be presented in a language that is supported by Alexa and the language should match the language
>  for which the skill is created. If the skill supports multiple languages, then the description for each language has to be presented in the relevant language. Alexa
> 's users should be able to read the skill's description in the locale they intend to use the skill.